File: practice23/1.cpp

//https://www.nowcoder.com/practice/44da6966beb449d383f62b12e8df6317?tpId=98&tqId=32853&tPage=2&rp=2&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
#include<string>
using namespace std;

int main()
{
	string str;
	char ch;
	int x;
	while (cin >> str)
	{
		ch = str[0];
		x = 0;
		for (int i = 0; i < str.size(); i++)
		{
			if (str[i] == ch)
			{
				x++;
			}
			else
			{
				cout << x << ch;
				ch = str[i];
				x = 1;
			}
		}
		cout << x << ch << endl;
	}
	system("pause");
	return 0;
}

File: practice30/1.java

//https://www.nowcoder.com/practice/ce5b35929ab84e3a806886d9667be00a?tpId=90&tqId=30790&tPage=1&rp=1&ru=/ta/2018test&qru=/ta/2018test/question-ranking
import java.util.Scanner;

public class Main {
    public static void main(String args[]){
        Scanner scanner=new Scanner(System.in);
        int n=scanner.nextInt();
        int t=scanner.nextInt();
        int a=scanner.nextInt();
        int count=0;
        if(t<=a){
            count=count+t;
            count=count+n-a;
        }
        else{
            count=count+a;
            count=count+n-t;
        }
        System.out.println(count);
    }
}
